### PR TITLE
fix: avoid double delimiters

### PR DESCRIPTION
**Pull request by slin-sap**
Fix for : #288

- Forgot to remove ? when building the sap-client parameter. This ? is duplicate when adding ? to the full query param. 
- Use node built-in URLSearchParams instead of using join('&') to build the full query param.

### CODE REVIEW

**Review by devinea — DISMISSED** [review]
Lgtm

File: packages/fiori-freestyle-writer/src/packageConfig.ts [hunk 1]
```diff
@@ -28,10 +28,20 @@ export function getPackageJsonTasks({
...
+        searchParamList.push([`sap-client`, `${sapClient}`]);
+    }
+    searchParamList.push(['sap-ui-xx-viewCache', 'false']);
+
+    let searchParam = new URLSearchParams(searchParamList).toString();
+    searchParam = searchParam ? `?${searchParam}` : '';
+    // Build fragment identifier part of url
+    const harshFragment = flpAppId ? `#${flpAppId}` : '';
+    // Full parameter section composed by search param and fragment identifier
+    const params = `${searchParam}${harshFragment}`;
```
Comment (ullasholla) [inline packages/fiori-freestyle-writer/src/packageConfig.ts:43]: `harshFragment` -> `hashFragment` or even just `fragment` or `urlFragment`?

**Review by ullasholla — CHANGES_REQUESTED** [review]
Looks good, other than that typo. Cheers.

**Comment by ullasholla** [comment]
Looks good, @slin-sap, thank you. Thanks for using `URLSearchParams` to build up the search string.

**Review by ullasholla — APPROVED** [review]

**Review by devinea — APPROVED** [review]

**Review by IainSAP — COMMENTED** [review]
Thx @slin-sap

File: packages/fiori-freestyle-writer/src/packageConfig.ts [hunk 2]
```diff
@@ -28,10 +28,20 @@ export function getPackageJsonTasks({
...
-    const params = `${urlParam}${flpAppId ? `#${flpAppId}` : ''}`;
+    // Build search param part of preview launch url
+    const searchParamList = [];
+    if (sapClient) {
+        searchParamList.push([`sap-client`, `${sapClient}`]);
+    }
+    searchParamList.push(['sap-ui-xx-viewCache', 'false']);
+
+    let searchParam = new URLSearchParams(searchParamList).toString();
+    searchParam = searchParam ? `?${searchParam}` : '';
```
Comment (IainSAP) [inline packages/fiori-freestyle-writer/src/packageConfig.ts:39]: Unnecessary condition since `searchParamList` always has a value.

**Pull request merged**